WooCommerce/Classes/ViewRelated/Coupons/CouponManagementViewController.swift

```diff
@@ -1,25 +1,205 @@
+            displayPlaceholderCoupons()
+        case .coupons:
+            tableView?.reloadData()
+        case .empty:
+            displayNoResultsOverlay()
+        default:
+            break
+        }
     }
 }
 
+
 // MARK: - View Configuration
 //
 private extension CouponManagementViewController {
 
     func configureNavigation() {
         title = Localization.title
     }
+
+    func configureTableView() {
+        registerTableViewCells()
+        tableView?.dataSource = self
+        tableView?.estimatedRowHeight = Constants.estimatedRowHeight
+        tableView?.rowHeight = UITableView.automaticDimension
+    }
+
+    func registerTableViewCells() {
+        tableView?.registerNib(for: TitleBodyTableViewCell.self)
+    }
 }
 
+
+// MARK: - Placeholder cells
+//
+extension CouponManagementViewController {
+    /// Renders the Placeholder Coupons
+    ///
+    func displayPlaceholderCoupons() {
+        let options = GhostOptions(displaysSectionHeader: false,
+                                   reuseIdentifier: TitleBodyTableViewCell.reuseIdentifier,
+                                   rowsPerSection: Constants.placeholderRowsPerSection)
+        tableView?.displayGhostContent(options: options,
+                                       style: .wooDefaultGhostStyle)
+    }
+
+    /// Removes the Placeholder Coupons
+    ///
+    func removePlaceholderCoupons() {
+        tableView?.removeGhostContent()
+    }
+}
+
+
+// MARK: - Empty state view controller
+//
+extension CouponManagementViewController {
+    /// Displays the overlay when there are no results.
+    ///
+    func displayNoResultsOverlay() {
+        let emptyStateViewController = EmptyStateViewController(style: .list)
+        let config = EmptyStateViewController.Config.withButton(
+            message: .init(string: Localization.emptyStateMessage),
+            image: .errorImage,
+            details: Localization.emptyStateDetails,
+            buttonTitle: "") { _ in }
+
+        displayEmptyStateViewController(emptyStateViewController)
+        emptyStateViewController.configure(config)
+    }
+
+    /// Shows the EmptyStateViewController as a child view controller.
+    ///
+    func displayEmptyStateViewController(_ emptyStateViewController: UIViewController) {
+        self.emptyStateViewController = emptyStateViewController
+        addChild(emptyStateViewController)
+
+        emptyStateViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(emptyStateViewController.view)
+
+        NSLayoutConstraint.activate([
+            emptyStateViewController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            emptyStateViewController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            emptyStateViewController.view.topAnchor.constraint(equalTo: view.topAnchor),
+            emptyStateViewController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
```

Comment:
nit: there is `pinSubviewToAllEdges` helper that can simplify this

Reply:
Thanks!